src/olympia/addons/tasks.py

```diff
@@ -289,14 +289,14 @@ def update_addon_hotness(averages):
 def update_addon_weekly_downloads(data):
     log.info('[%s] Updating add-ons weekly downloads.', len(data))
 
-    for addon_guid, count in data:
+    for hashed_guid, count in data:
         try:
-            addon = Addon.objects.get(guid=addon_guid)
-        except Addon.DoesNotExist:
+            addon = AddonGUID.objects.get(hashed_guid=hashed_guid).addon
```

Comment:
You'll need some extra filtering on this - `AddonGUID` contains _all_ instances of a `GUID` being used and you just want the most recent addon instance rather than previous deleted ones.

I guess for _downloads_ you can just filter out deleted addons - all of the addons might have been deleted - but deleted addons don't get downloads.  (unlike ADU when you do want to keep updating the stats for admin purposes).

Reply:
`.get()` would have failed when there were multiple AddonGUID with the same guid (hashed_guid)

Reply:
Do I simply have to exclude `addon=None`?

Reply:
I am confused, how can we retrieve the last/current add-on?

Reply:
The easiest way is probably just to rely on the Addon objects manager (like now) that filters out non-deleted addons.  So it'd be `Addon.objects.get(addonguid__hashed_guid=hashed_guid)`